src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -16,15 +16,14 @@ namespace Stratis.Bitcoin.Features.BlockStore
     /// <summary>
     /// <see cref="IBlockRepository"/> is the interface to all the logics interacting with the blocks stored in the database.
     /// </summary>
-    /// <seealso cref="System.IDisposable" />
-    public interface IBlockRepository : IBlockStore, IDisposable
+    public interface IBlockRepository : IBlockStore
     {
         /// <summary>
         /// Persist the next block hash and insert new blocks into the database.
         /// </summary>
-        /// <param name="nextBlockHash">next block hash</param>
-        /// <param name="blocks">blocks to be inserted</param>
-        Task PutAsync(uint256 nextBlockHash, List<Block> blocks);
+        /// <param name="newTip">New repository's tip.</param>
```

Comment:
You could make it consistent with the comment of the similar parameter in delete